rhalphalib/sample.py

```diff
@@ -105,20 +105,29 @@ def __init__(self, name, sampletype, template):
             (for the others, the observable name is taken from the x axis name)
         '''
         super(TemplateSample, self).__init__(name, sampletype)
-        sumw, binning, obs_name = _to_numpy(template)
+        sumw2 = None
+        try:
+            sumw, binning, obs_name, sumw2 = _to_numpy(template, read_sumw2=True)
+        except ValueError:
```

Comment:
Looks like `_to_numpy` can also return IndexError when the tuple is not long enough. Probably `_to_numpy` should return `ValueError` if `read_sumw2` is true and the tuple is not long enough to have that info. Or you can `except ValueError, IndexError`

Reply:
Took your first suggestion